ngrok/src/session.rs

```diff
@@ -737,19 +785,36 @@ async fn accept_incoming(mut incoming: IncomingStreams, inner: Arc<ArcSwap<Sessi
     let error: AcceptError = loop {
         if let Err(error) = accept_one(&mut incoming, &inner).await {
             debug!(%error, "failed to accept stream, attempting reconnect");
-            let reconnect = Retry::spawn(ExponentialBackoff::from_millis(50), || {
-                try_reconnect(inner.clone())
-            });
+            // This is gross, but should perform fine. Couple of notes:
+            // * Mutex so that both the action and condition can share access to
+            //   `error`. Realistically, the lock calls should be non-concurrent,
+            //   but Rust can't prove that.
+            // * Not setting the error in the action because then a a reference
+            //   to a FnMut closure would escape via the returned Future, which is
+            //   a no-no.
+            let error = parking_lot::Mutex::new(Some(error));
```

Comment:
Why use `parking_lot` Mutex instead of `tokio`? Faster speed of reconnect where contention shouldn't actually be an issue?

Reply:
It's my go-to synchronous mutex - the one in `tokio` is optimized for async usecases where you have to do stuff like hold it across a yield point.